app/lib/package/api_export/export_api_to_bucket.dart

```diff
@@ -78,6 +81,83 @@ class ApiExporter {
         .write(await searchBackend.getPackageNameCompletionData());
   }
 
+  Future<void> fullSync() async {
+    final invisiblePackageNames = await dbService
+        .query<ModeratedPackage>()
+        .run()
+        .map((mp) => mp.name!)
+        .toSet();
+
+    final allPackageNames = <String>{};
+    final packageQuery = dbService.query<Package>();
+    await packageQuery.run().parallelForEach(_concurrency, (pkg) async {
+      final name = pkg.name!;
+      if (pkg.isNotVisible) {
+        invisiblePackageNames.add(name);
+        return;
+      }
+      allPackageNames.add(name);
+
+      // TODO: Consider retries around all this logic
+      await syncPackage(name);
```

Comment:
I think `ExportedApi.garbageCollect(allPackageNames)` will delete any moderated packages.

The only case where it won't is if the files in question are modified recently, then it won't delete them. But in this case it will delete them eventually.

And our admin action would delete them, so it's only in the odd corner cases where the GC is necessary to delete moderated packages.

Reply:
> The only case where it won't is if the files in question are modified recently, then it won't delete them. But in this case it will delete them eventually.

The current moderation actions use the following call to update the public bucket:
```
      await packageBackend.tarballStorage.updatePublicArchiveBucket(
        package: package,
        ageCheckThreshold: Duration.zero,
        deleteIfOlder: Duration.zero,
      );
```
I imagine we only want to have a single method for this, and a similar parameter option would be nice here too.

Reply:
(There is no separate delete operation for the files, it is reusing the sync/update method with modified thresholds.)